src/Azure/Queues/AzureServiceBusQueue.cs

```diff
@@ -1,79 +1,149 @@
 
-            if (!_namespaceManager.QueueExists(_queueName)) {
-                _queueDescription = new QueueDescription(_queueName) {
-                    MaxDeliveryCount = retries + 1,
-                    LockDuration = _workItemTimeout
-                };
-                _namespaceManager.CreateQueue(_queueDescription);
-            } else {
-                _queueDescription = _namespaceManager.GetQueue(_queueName);
-
-                bool changes = false;
-
-                int newMaxDeliveryCount = retries + 1;
-                if (_queueDescription.MaxDeliveryCount != newMaxDeliveryCount) {
-                    _queueDescription.MaxDeliveryCount = newMaxDeliveryCount;
-                    changes = true;
-                }
+            // maintain previous functionality: go ahead and synchronously create queue and client
+            var queueDescription = EnsureQueueExistsAsync().Result;
+
+            _clientFactory = MemoizeClientFactory(connectionString, retryPolicy);
+            var ignoreResult = _clientFactory(queueDescription.Path);
+        }
+
+        private AzureServiceBusQueue(string queueName, NamespaceManager namespaceManager, Func<Task<QueueDescription>> queueFactory, Func<string, QueueClient> clientFactory, ISerializer serializer, IEnumerable<IQueueBehavior<T>> behaviors, ILoggerFactory loggerFactory) : base(serializer, behaviors, loggerFactory) {
+            _queueName = queueName;
+            _namespaceManager = namespaceManager;
+            _createQueue = queueFactory;
+            _clientFactory = clientFactory;
+        }
+
+        public class Factory {
+            private readonly string _connectionString;
+            private string _queueName;
+            private int _retries;
+            private TimeSpan _workItemTimeout;
+            private bool _shouldRecreate;
+            private bool _lazyConnect;
+            private RetryPolicy _retryPolicy;
+            private ISerializer _serializer;
+            private IEnumerable<IQueueBehavior<T>> _behaviors;
+            private ILoggerFactory _loggerFactory;
+
+            public Factory(string connectionString, string queueName = null, int retries = 2, TimeSpan? workItemTimeout = null, bool shouldRecreate = false, bool lazyConnect = true, RetryPolicy retryPolicy = null, ISerializer serializer = null, IEnumerable<IQueueBehavior<T>> behaviors = null, ILoggerFactory loggerFactory = null) {
+                _connectionString = connectionString;
+                Queue(queueName);
+                _retries = retries;
+                Timeout(workItemTimeout ?? TimeSpan.FromMinutes(5));
+                _shouldRecreate = shouldRecreate;
+                _lazyConnect = lazyConnect;
+                _retryPolicy = retryPolicy;
+                _serializer = serializer;
+                _behaviors = behaviors;
+                _loggerFactory = loggerFactory;
+            }
+
+            public Factory Queue(string queueName) {
```

Comment:
Yeah, either works for me. I agree the properties version looks better. I'll change it.